modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/TypeWrapper.java

```diff
@@ -193,7 +193,7 @@ public boolean isSameType(Type type) {
 
   @Override
   public boolean isSameType(Class<?> clazz) {
-    return aClass.equals(clazz);
+    return aClass != null && aClass.equals(clazz);
```

Comment:
how could this ever happen? How would somebody create a type wrapper around a null class?

Reply:
It happens when the type is `?`. If I don't make this check I'll get NPE when doing `generics.get(0).getConcreteType().isSameType(Void.class`

Reply:
but then shouldn't the same check be added everywhere? Why not using `Object.class` so that it acts as a NullObject pattern?

Reply:
The same check it's now in the others methods:
`public boolean isAssignableTo(Class<?> clazz)` and `public boolean isAssignableFrom(Class<?> clazz)`
It looks like someone forgot to add it to `public boolean isSameType(Class<?> clazz)`

I guess it's possible to assign `Object.class` to `aClass` if the constructor gets a `null`, do you think it's a better solution?

Reply:
the check is indeed present in those other methods, but it's not present in all the rest, meaning that there's a potential NPE if any other change starts invoking a method that's not being invoked right now. I think we should completely prevent the NPE situation altogether and using a NullObject is both the easiest and safest way of doing that.